### ISSUE TITLE
rule: inline-no-box-sizing

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#130]
## Rule ID
`inline-no-box-sizing`

## What does this rule detect?
`box-sizing` has no effect on non-replaced inline elements because `width`/`height` do not apply. Since `box-sizing` only affects how width/height are calculated, it's meaningless when those properties are ignored.

## Example: should warn
```html
<span style="display: inline; box-sizing: border-box;">text</span>
```

## Example: should NOT warn
```html
<div style="box-sizing: border-box;">content</div>
```

## Related CSS Properties
`display`, `box-sizing`

## References
- [CSS Box Sizing Level 3](https://drafts.csswg.org/css-sizing/#box-sizing): affects width/height calculation
- [Firefox inactive-property-helper.js](https://searchfox.org/mozilla-central/source/devtools/server/actors/utils/inactive-property-helper.js) — detects this pattern
- Needs `requiredInlineProperties` guard to avoid false positives from universal `* { box-sizing: border-box }` resets
